specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2018-07-10/service.json

```diff
@@ -15724,6 +15724,26 @@
       },
       "x-ms-discriminator-value": "New"
     },
+    "NewRecoveryVirtualNetwork": {
+      "description": "Recovery virtual network input to create new virtual network from given source network.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/RecoveryVirtualNetworkCustomDetails"
+        }
+      ],
+      "properties": {
+        "recoveryVirtualNetworkResourceGroupName": {
```

Comment:
I assume you will be creating this vnet via S2S call from your RP to Network RP - which means
- your service role definition needs to be updated by NRP team with Microsoft.Network/virtualNetworks/write permission (if you don't already have it)
- the subscription in which you create this vnet must have your RP registered - if the subscription you create the vnet in is the target VM's subscription then I assume you assert that the VM's sub == the sub which contains the recovery service's vault

Reply:
Service already has option for network creation through different flow. This update would merely plug into existing workflow. So all the necessary permissions are already in place.

Reply:
thanks for confirming